### PR TITLE
DPatch: def parameter updated to accept true labels

### PR DESCRIPTION
**Pull request by yuki2222**
# Description

This pull request updates DPatch so that it accepts the true labels `y`.

Fixes # (issue)

## Type of change

Please check all relevant options.

- [x] Improvement (non-breaking)
- [ ] Bug fix (non-breaking)
- [ ] New feature (non-breaking)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Testing

Please describe the tests that you ran to verify your changes. Consider listing any relevant details of your test configuration.

- [x] Test A
- [x] Test B

**Test Configuration**:
- OS: Ubuntu 18.04.5 LTS
- Python version: 3.7.13
- ART version or commit number: 2be96ee580f15bd6aacd64f2320cc8ccf47fe891 70d6ed29c1b91f17513abb213aff8a2c1491da96
- TensorFlow / Keras / PyTorch / MXNet version: 2.8.2 / 2.8.0 / 1.12.0

# Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: art/attacks/evasion/dpatch.py [hunk 1]
```diff
@@ -107,7 +107,12 @@ def generate(  # pylint: disable=W0221
         Generate DPatch.
 
         :param x: Sample images.
-        :param y: Target labels for object detector.
+        :param y: True labels of type `List[Dict[np.ndarray]]` for untargeted attack, one dictionary per input image. The
+                  keys and values of the dictionary are:
+
```
Comment (beat-buesser) [inline art/attacks/evasion/dpatch.py:112]:
```suggestion
        :param y: True labels of type `List[Dict[np.ndarray]]` for untargeted attack, one dictionary per input image.
                  The keys and values of the dictionary are:

```

**Review by beat-buesser — CHANGES_REQUESTED** [review]
Hi @yuki2222 Thank you very much for your first PR to ART! Can I ask you to make two additional changes:

- remove lines 74-76 in `tests/attacks/evasion/test_dpatch.py` to remove the now obsolete check for raising ValueError if `y` is not None
- commit the changes suggested above for the new docstring, one of its lines is too long

**Comment by yuki2222** [comment]
Thank you!!!
